docs/sdk/api/sdkAuth.md

```diff
@@ -244,3 +251,50 @@ await authClient.introspectToken('invalid_token')
 //   "active":false
 // }
 ```
+
+### Token Provider
+
+Token provider is a special class which watches over `access_token` and refreshes it using `sdkAuth.refreshTokenFlow()` method if it is expired.
```

Comment:
@junajan it says here that it refreshes the token using the refreshTokenFlow. But how does it refresh it if it doesn't have a refresh token (i.e. it's client credentials flow)?

Reply:
Hi @lauraluiz .. this implementation as it is now will not generate new access token if the `refresh_token` is not provided or if it is expired.

Reply:
Hi @lauraluiz .. please check my latest update with additional functionality f62d2cd